.gitlab-ci.yml

```diff
@@ -0,0 +1,113 @@
+          cmake $cupla_DIR/example/CUDASamples/matrixMul/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+          && make -j
+          && ./matrixMul -wA=64 -wB=64 -hA=64 -hB=64
+          && rm -r * ;
+        fi
+        && echo "###################################################"
+        && echo "# Example Async API (adapted original)"
+        && echo "###################################################"
+        && echo "can not run with CPU_B_SEQ_T_SEQ due to missing elements layer in original SDK example"
+        && echo "CPU_B_SEQ_T_OMP2/THREADS too many threads necessary (512)"
+        && if [[ $CMAKE_FLAGS =~ -*DALPAKA_ACC_GPU_CUDA_ENABLE=ON.* ]]; then
+          cmake $cupla_DIR/example/CUDASamples/asyncAPI/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+          && make -j
+          && ./asyncAPI
+          && rm -r * ;
+        fi
+        && echo "###################################################"
+        && echo "# Example Async API (added elements layer)"
+        && echo "###################################################"
+        && cmake $cupla_DIR/example/CUDASamples/asyncAPI_tuned/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+        && make -j
+        && ./asyncAPI_tuned
+        && rm -r *
+        && echo "###################################################"
+        && echo "Example vectorAdd (added elements layer)"
+        && echo "###################################################"
+        && cmake $cupla_DIR/example/CUDASamples/vectorAdd/ $CMAKE_FLAGS -DCMAKE_BUILD_TYPE=Release
+        && make -j
+        && ./vectorAdd 100000
+        && rm -r * ;
+      done
+
+.base_gcc:
+  variables:
+    GIT_SUBMODULE_STRATEGY: normal
+    CXX: g++
+    CC: gcc
+    ALPAKA_ACCS: "-DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON
+                  -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=ON
+                  -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE=ON"
+                  # -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE=ON
+  before_script:
+    - echo "number of processor threads $(nproc)"
+    - $CXX --version
+  extends: .base_job
+  # the rocm runner is used to get a multi-core CPU for the tests
+  tags:
+    - amd
+    - rocm
+
+.base_clang:
+  variables:
+    GIT_SUBMODULE_STRATEGY: normal
+    CXX: clang++
+    CC: clang
+    ALPAKA_ACCS: "-DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON
+                  -DALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE=ON"
+                  # -DALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE=ON
+                  # -DALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE=ON
+  before_script:
+    - echo "number of processor threads $(nproc)"
+    - $CXX --version
+  extends: .base_job
+  # the rocm runner is used to get a multi-core CPU for the tests
+  tags:
+    - amd
+    - rocm
+
+.base_cuda:
+  variables:
+    GIT_SUBMODULE_STRATEGY: normal
+    CXX: g++
+    CC: gcc
+    ALPAKA_ACCS: "-DALPAKA_ACC_GPU_CUDA_ENABLE=ON"
+  before_script:
+    - nvidia-smi
+    - nvcc --version
+    - gcc --version
```

Comment:
`gcc --version` to `$CXX --version` (in general base job section?)?

Reply:
fixed